### DISCUSSION TITLE
Always cache files and save them

### DISCUSSION THREAD

**Discussion by dereference23** [community/community#26467] [Actions]
<p>Hello! I need cache to be restored and saved again every time the job ends. How to do this?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/ghost6482">@ghost6482</a> ,</p>
<p>When <a href="https://github.com/actions/cache/blob/f00dedfa6c708c51f4e4296f092458a4a891bddd/action.yml#L15" rel="nofollow noopener">cache-hit</a> value is false, the cache will be restored and saved again.  Code sample as below:</p>
<pre><code>- name: Cache node modules
      id: cachetest
      uses: actions/cache@v1
      env:
        cache-name: cache-node-modules
      with:
        path: npm # npm cache files are stored in `npm` on Linux/macOS
        key: ${{ runner.os }}-build-${{ env.cache-name }}-${ github.sha }}-${{ hashFiles('**/package.json') }} # Add githu.sha to make the key different each time
        restore-keys: |
          ${{ runner.os }}-build-${{ env.cache-name }}-
          ${{ runner.os }}-build-
          ${{ runner.os }}-
</code></pre>
<p>I added ${{githu.sha}} to make sure the key is different for each commit, which will cause ‘cache-hit’ value is ‘false’.</p>
<p>Cache restored: <a href="https://github.com/weide-zhou/github-actions-for-packages/runs/620991567?check_suite_focus=true#step:3:13" rel="nofollow noopener">https://github.com/weide-zhou/github-actions-for-packages/runs/620991567?check_suite_focus=true#step:3:13</a></p>
<p>Cache saved: <a href="https://github.com/weide-zhou/github-actions-for-packages/runs/620991567?check_suite_focus=true#step:9:3" rel="nofollow noopener">https://github.com/weide-zhou/github-actions-for-packages/runs/620991567?check_suite_focus=true#step:9:3</a></p>
<p>Hope it’s clear and helpful.</p>
